lit_tests/kernel/wave/attention/extend_attention.py

```diff
@@ -353,16 +354,18 @@ def test_causal_extend_attention_32x32x8():
 
     # CHECK-COUNT-2:            gpu.shuffle xor {{.*}}
     # CHECK-COUNT-8:            amdgpu.mfma
+    # CHECK:                %[[ALLOC3:.*]] = memref.view %[[ALLOC0]][%[[C0]]][] : memref<8960xi8, #gpu.address_space<workgroup>> to memref<1x64x36xf16, #gpu.address_space<workgroup>>
+    # CHECK:                %[[ALLOC4:.*]] = memref.view %[[ALLOC0]][%[[C4608]]][] : memref<8960xi8, #gpu.address_space<workgroup>> to memref<1x32x68xf16, #gpu.address_space<workgroup>>
     # CHECK-COUNT-8:        vector.maskedload
     # CHECK:                amdgpu.lds_barrier
     # CHECK-NOT:            amdgpu.lds_barrier
     # CHECK:                stream.binding.subspan %{{.*}}[%{{.*}}] : !stream.binding -> memref<?x4x64xf16, strided<[256, 64, 1], offset: ?>>
     # CHECK:                stream.binding.subspan %{{.*}}[%{{.*}}] : !stream.binding -> memref<?x4x64xf16, strided<[256, 64, 1], offset: ?>>
     # CHECK:                scf.for
     # CHECK-COUNT-1:            vector.maskedload
-    # CHECK-COUNT-1:            vector.store %{{.*}}, %[[ALLOC2]]
-    # CHECK-COUNT-32:           memref.load %{{.*}}
-    # CHECK-COUNT-8:            vector.load %[[ALLOC2]]
+    # CHECK-COUNT-2:            vector.store %{{.*}}, %[[ALLOC4]]
+    # CHECK-COUNT-8:            vector.maskedload %{{.*}}
```

Comment:
So is the reason we get a speedup on attention the fact that we have 8 masked loads of size 4 vs 32 loads of size 1?

Reply:
Without `in-thread-transpose` `global-to-shared-gather` was generating a lot of 1-element reads from shared